inspire_utils/name.py

```diff
@@ -81,9 +83,13 @@ def __init__(self, name, constants=None):
         if isinstance(name, HumanName):
             self._parsed_name = name
         else:
+            self._parsed_name = HumanName(name, constants=constants)
             self._parsed_name = HumanName(name, constants=constants)
             self._parsed_name.capitalize()
 
+        if len(self.first_list) == 1 and '.' not in self.first and ',' not in name:
```

Comment:
why the `len` check now? I don't remember it being there previously.

Reply:
The thing is that `self.first` is only one first name, so names like `James K. Smith` also were marked as True what's wrong

Reply:
how is that possible, if it contains a `.`? does `self.first` exclude middle names?

Reply:
Yes, it's only first name (one string, not a list)

Reply:
but it's not correct: `John David Smith` might be a last name too.

Reply:
So it is wrong now in query parser. I'll change it to create `full_name` query for those cases too.